File: benchmark/corpus/12_indirect_authority/a_gains_net.py
# Reads a price from the shop's feed and prints it through a formatting library.
import sys
import urllib.request

import pricing

feed = sys.stdin.readline().strip()
try:
    body = urllib.request.urlopen(feed).read()
    print(pricing.price_line("chai", len(body)))
except Exception as e:
    print("no price feed: " + str(e))

File: benchmark/corpus/12_indirect_authority/a_gains_net.js
// Reads a price from the shop's feed and prints it through a formatting library.
import { priceLine } from "./pricing.js";

const feed = (await new Response(Deno.stdin.readable).text()).trim();
try {
  const body = await (await fetch(feed)).text();
  console.log(priceLine("chai", body.length));
} catch (e) {
  console.log("no price feed: " + e.message);
}
